docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,140 @@
+# MeshService migration
+
+* Status: accepted
+
+## Context and Problem Statement
+
+This MADR addresses the switch from not having MeshServices at all to MeshServices
+being automatically created, either from `Services` in Kubernetes or from
+`Dataplanes` in Universal.
+
+When upgrading from 2.8 to 2.9, nothing at the data plane level should change.
+But MeshService behaves differently in several different ways.
+
+### Policy
+
+If for example we have a policy targeted at:
+
+```
+spec:
+  targetRef:
+    kind: MeshService
+    name: demo-app_kuma-demo_svc_5000
+```
+
+when switching over to MeshService, this policy will no longer apply.
+
+### Local traffic
+
+Because the `MeshServices` we generate have selectors equivalent to the `Service`,
+they will select the same `Dataplanes` that previously had the `kuma.io/service`
+corresponding to the `Service` and thus, absent any policies, traffic flows to
+the same endpoints and Envoy has the same cluster configuration.
+
+### Cross-zone traffic
+
+One of the fundamental changes with `MeshService` is that traffic to a service
+is no longer load balanced across all zones.
+
+We need to take measures to prevent this change from happening on upgrade.
+
+In particular with Kubernetes, it's important to keep in mind that after
+MeshService is enabled we no longer generate our own VIP.
+So any requests to MeshService hostnames go to the same
+outbound listener as with requests to Kubernetes DNS names, the ClusterIP.
+
+#### Status quo
+
+- On Kubernetes there are two outbound listeners:
+  - Traffic to the Kubernetes DNS name/ClusterIP is load-balanced cross-zone
+  - Traffic to the `<kuma.io/service>.mesh`/VIP is load-balanced cross-zone
+
+#### `MeshService`
+
+- We now have one outbound listener:
+  - Traffic to the generated hostname of the `MeshService` goes through the Cluster IP
+  - This cluster IP is no longer cross-zone load balanced
+
+## Considered Options
+
+* Policies applied to `demo-app_kuma-demo_svc_5000` should also apply to
```

Comment:
indefinitely or for some transition period and then it's going to get removed?

Reply:
as long as we have `kuma.io/service` I suppose